File: 2nd_try/0001_Two_Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        past = {}

        for i in range(len(nums)):
            if target-nums[i] in past:
                return [past[target-nums[i]],i]
            else:
                past[nums[i]] = i

File: Java_solution/0001_Two_Sum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> exist = new HashMap<>();
        
        for (int i=0;i<nums.length;i++){
            if (exist.containsKey(target-nums[i])){
                return new int[]{exist.get(target-nums[i]),i};
            }
            exist.put(nums[i],i);
        }
        return new int[]{};
        
    }
}